obspy/signal/polarization.py

```diff
@@ -86,48 +87,482 @@ def eigval(datax, datay, dataz, fk, normf=1):
+        diffstart = slatest - stream[i].stats.starttime
+        frac, ddummy = math.modf(diffstart)
+        spoint[i] = int(ddummy)
+        if frac > stream[i].stats.delta * 0.25:
+            msg = "Difference in start times exceeds 25% of samp rate"
+            warnings.warn(msg)
+        spoint[i] += offset
+        diffend = stream[i].stats.endtime - eearliest
+        frac, ddummy = math.modf(diffend)
+        epoint[i] = int(ddummy)
+        epoint[i] += negoffset
+
+    return spoint, epoint
+
+
+def polarization_analysis(stream, win_len, win_frac, frqlow, frqhigh, stime,
+                          etime, verbose=False, timestamp="mlabday",
+                          method="pm", var_noise=0.0):
+    """
+    Method carrying out polarization analysis with the Flinn, Jurkevics,
+    ParticleMotion, or Vidale algorithm.
+
+    :param stream: 3 component input data.
+    :type stream: :class:`~obspy.core.stream.Stream`
+    :param win_len: Sliding window length in seconds.
+    :type win_len: float
+    :param win_frac: Fraction of sliding window to use for step.
+    :type win_frac: float
+    :param var_noise: resembles a sphere of noise in PM where the 3C is
+        excluded
+    :type var_noise: float
+    :param frqlow: lower frequency for PM
+    :type frqlow: float
+    :param frqhigh: higher frequency for PM
+    :type frqhigh: float
+    :param stime: Start time of interest
+    :type stime: :class:`obspy.core.utcdatetime.UTCDateTime`
+    :param etime: End time of interest
+    :type etime: :class:`obspy.core.utcdatetime.UTCDateTime`
+    :param timestamp: valid values: ``"julsec"`` and ``"mlabday"``;
+        ``"julsec"`` returns the timestamp in seconds since
+        ``1970-01-01T00:00:00``, ``"mlabday"`` returns the timestamp in days
+        (decimals represent hours, minutes and seconds) since
+        ``0001-01-01T00:00:00`` as needed for matplotlib date plotting (see
+        e.g. matplotlibs num2date)
+    :type timestamp: str
+    :param method: the method to use. one of ``"pm"``, ``"flinn"`` or
+        ``"vidale"``.
+    :type method: str
+    :returns: Dictionary with azimuth, incidence angle, errors,
+        rectilinearity, planarity, and/or ellipticity (the returned values
+        depend on the used method).
+    """
+    if method.lower() not in ["pm", "flinn", "vidale"]:
+        msg = "Invalid method ('%s')" % method
+        raise ValueError(msg)
+
+    res = []
+
+    # check that sampling rates do not vary
+    fs = stream[0].stats.sampling_rate
+    if len(stream) != len(stream.select(sampling_rate=fs)):
+        msg = "in array sampling rates of traces in stream are not equal"
+        raise ValueError(msg)
+
+    if verbose:
+        print("stream contains following traces:")
+        print(stream)
+        print("stime = " + str(stime) + ", etime = " + str(etime))
+
+    # offset of arrays
+    spoint, _epoint = get_s_point(stream, stime, etime)
```

Comment:
Is this not the only caller of `get_s_point`? Why bother calculating `epoint` then?